nodejs/eks/cluster.ts

```diff
@@ -483,14 +485,55 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
             return mappingYaml;
         });
     }
-    const eksNodeAccess = new k8s.core.v1.ConfigMap(`${name}-nodeAccess`, {
-        apiVersion: "v1",
-        metadata: {
-            name: `aws-auth`,
-            namespace: "kube-system",
-        },
-        data: nodeAccessData,
-    }, { parent: parent, provider: provider });
+    /*
+    * We must kubectl apply the aws-auth configmap since the creation of an
+    * AWS managed node group automatically triggers AWS to create the aws-auth
+    * configmap for us.
+    *
+    * For managed nodegroups:
+    *   To properly create aws-auth, we need to kubectl apply the configmap
+    *   instead of kubectl create it, if not it will fail with an existing
+    *   resource error message.
+    *
+    * For self-managed nodegroups:
+    *   We continue to always create the aws-auth configmap.
```

Comment:
I don't see this implemented anywhere - we seem to have just deleted the `new k8s.core.v1.ConfigMap`.

Won't that be a breaking change for existing deployments? (do we have any update testing that would verify this doesn't break things?)

Reply:
Yeah, it looks like this change moves this CM out of Pulumi's management for both cases. I expect that we'd want to keep the existing way for non-managed node groups.

Reply:
The pulumi managed ConfigMap was replaced with the `kubectl apply` - they both produce the same configmap result in the cluster, but it is now no longer under Pulumi management.

We must do this to satisfy both configmap scenarios when using self-managed nodegroups vs AWS managed.

This is not a breaking change as the CI runs with `RunUpdateTest` on all existing tests to update to this PR and all tests are green.

>  I expect that we'd want to keep the existing way for non-managed node groups

We cannot deduce this from the cluster `createCore()` (the orig location of the configmap creation), as the user has a couple of ways of creating self-managed node groups and using the new `createManagedNodeGroup()` API. Naturally it felt easier to replace this here with `kubectl apply`

Reply:
If I understand what will happen during an upgrade to a new version of this package when using existing node groups:
1. During the update, we will now do a `kubectl apply` which will in principle update the existing ConfigMap and make no changes.
2. At the end of the update, we will destroy the `${name}-nodeAccess` ConfigMap as it no longer exists in the Pulumi program, this will blow away the changes that were just made above.

Are you sure that the above does not happen?  If it does not - what does in practice happen during this update? (and how do we avoid the removal of the `new k8s.core.v1.ConfigMap` from destroying the ConfigMap?

Reply:
I tried this on existing self-managed nodegroups, and you are correct that deleting the nodeAccess pulumi-managed CM removes the `aws-auth` file configmap after an update to the latest version of this PR.